Loan.ts

```diff
@@ -0,0 +1,293 @@
+  loanProductId:number,
+  loanProductName:string,
+  loanProductDescription:string,
+  loanPurposeId:number,
+  loanPurposeName:string,
+  loanOfficerId:number,
+  loanOfficerName:string,
+  loanType:LoanTypeInterface,
+  currency:LoanCurrencyInterface, 
+  principal:number,
+  termFrequency:number,
+  termPeriodFrequencyType:LoanTermPdFrequencyType ,
+  numberOfRepayments:number,
+  repaymentEvery:number,
+  repaymentFrequencyType:LoanRepaymentFrequencyType,
+  interestRatePerPeriod:number,
+  interestRateFrequencyType:LoanInterestRateFrequencyType ,
+  annualInterestRate:number,
+  amortizationType:LoanAmortizationType,
+  interestType:LoanInterestType ,
+  interestCalculationPeriodType:LoanInterestCalculationPeriodType,
+  transactionProcessingStrategyId:number,
+  timeline:LoanTimeline ,
+  summary:LoanSummary,    
+  linkedAccount:LoanLinkedAccount,
+  disbursementDetails:[LoanDisbursementDetails],
+  fixedEmiAmount:number,
+  maxOutstandingLoanBalance:number,
+  canDisburse:boolean,
+  emiAmountVariations:Array<number>,                //kindly review this type
+  inArrears:boolean,
+  isNPA:boolean,
+  overdueCharges:[LoanOverDueCharges]        
+}
+export interface LoanDisbursementDetails{
+  id:number,
+  expectedDisbursementDate:Array<number>,
+  principal:number,
+  approvedPrincipal:number
+}
+export interface LoanOverDueCharges{
+  id:number,
+  name:string,
+  active:boolean,
+  penalty:boolean,
+  currency:LoanCurrencyInterface ,
+  amount:number,
+  chargeTimeType:LoanChargeTimeType,
+  chargeAppliesTo:LoanChargeAppliesTo,
+  chargeCalculationType:LoanChargeCalType,
+  chargePaymentMode:LoanChargePaymentMode,
+  feeInterval:number,
+  feeFrequency:LoanFeeFrequency
+
+}  
+ 
+export interface LoanChargeTimeType{
+  id:number,
+  code:string,                      //"chargeTimeType.overdueInstallment",
+  value:string
+
+}    
+export interface LoanChargeAppliesTo{
+  id:number,
+  code:string,                              //"chargeAppliesTo.loan,
+  value:string
+}
+export interface LoanChargeCalType{
+  id:number,
+  code:string,                              //"chargeCalculationType.percent.of.amount",
+  value:string
+
+}
+export interface LoanChargePaymentMode{
+  id:number,
+  code:string,     //"chargepaymentmode.regular",
+  value:string                //"Regular"
+}
+export interface LoanFeeFrequency{
+  id:number,
+  code:string,    // "feeFrequencyperiodFrequencyType.weeks",
+  value:string //"Weeks"
+}
+export default class Loan {
+
+  fineract_obj: FineractAPI
+  constructor(fineract_obj: FineractAPI) {
+    this.fineract_obj = fineract_obj;
+  }
+
+  async list_loan(config: LoanListConfig): Promise<[LoanInterface]> {
+    let path = 'loans';
+    let response;
+    response = await this.fineract_obj.get(path, config);
+    return response;
+
+
+  }
+  async retrieve_loan(loanId: string, loan_retrive_config?: LoanRetrieveConfig): Promise<LoanRetrieveInterface> {
+    let retrieve_path = 'loans/${loanId}';
```

Comment:
string interpolation happens with `` and not ''. 
so the above code will become 

```javascript
let retrieve_path = `loans/${loanId}`;
```